File: src/main/java/eterea/core/api/rest/model/Comprobante.java
package eterea.core.api.rest.model;

import eterea.core.api.rest.kotlin.model.Auditable;
import eterea.core.api.rest.kotlin.model.Cuenta;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@Data
@Entity
@Table(name = "tiposcomprob")
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper=false)
public class Comprobante extends Auditable implements Serializable {

	private static final long serialVersionUID = 5004162435032433076L;

	@Id
	@Column(name = "codigo")
	private Integer comprobanteId;

	private String descripcion = "";

	@Column(name = "tco_neg_id")
	private Integer negocioId;

	private Integer modulo;
	private Byte stock = 0;
	private Byte rendicion = 0;

	@Column(name = "opago")
	private Byte ordenPago = 0;

	@Column(name = "aplicapend")
	private Byte aplicaPendiente = 0;

	@Column(name = "ctacte")
	private Byte cuentaCorriente = 0;

	private Byte debita = 0;
	private Byte iva = 0;
	private Byte ganancias = 0;
	private Byte aplicable = 0;

	@Column(name = "libroiva")
	private Byte libroIva = 0;

	@Column(name = "tipocomprob")
	private String letraComprobante;

	private Byte recibo = 0;
	private Byte contado = 0;
	private Byte transferencia = 0;
	private Byte imprime = 0;

	@Column(name = "cgoanulacion")
	private Integer comprobanteIdAnulacion;

	@Column(name = "cgocentroentrega")
	private Integer centroStockIdEntrega;

	@Column(name = "cgocentrorecibe")
	private Integer centroStockIdRecibe;

	@Column(name = "diasvigencia")
	private Integer diasVigencia;

	private Byte concepto = 0;
	private Byte personal = 0;
	private Byte comanda = 0;

	@Column(name = "tco_puntovta")
	private Integer puntoVenta = 0;

	@Column(name = "cgomozo")
	private Byte codigoMozo = 0;

	@Column(name = "tco_transferir")
	private Byte transferir = 0;

	@Column(name = "tco_cgocontable")
	private Long numeroCuenta;

	@Column(name = "tco_nivel")
	private Integer nivel = 0;

	@Column(name = "tco_fiscalizadora")
	private Byte fiscalizadora = 0;

	@Column(name = "tco_invisible")
	private Byte invisible = 0;

	@Column(name = "tco_tipoafip")
	private Integer comprobanteAfipId;

	@Column(name = "tco_factelect")
	private Byte facturaElectronica = 0;

	private Byte asociado = 0;

	@OneToOne(optional = true)
	@JoinColumn(name = "tco_cgocontable", insertable = false, updatable = false)
	private Cuenta cuenta;

}

File: src/main/java/eterea/core/service/kotlin/model/Comprobante.kt
package eterea.core.service.kotlin.model

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.json.JsonMapper
import eterea.core.service.model.Auditable
import eterea.core.service.tool.Jsonifier
import jakarta.persistence.*

@Entity
@Table(name = "tiposcomprob")
data class Comprobante(

    @Id
    @Column(name = "codigo")
    var comprobanteId: Int? = null,
    
    var descripcion: String = "",

    @Column(name = "tco_neg_id")
    var negocioId: Int? = null,

    var modulo: Int? = null,
    var stock: Byte = 0,
    var rendicion: Byte = 0,

    @Column(name = "opago")
    var ordenPago: Byte = 0,

    @Column(name = "aplicapend")
    var aplicaPendiente: Byte = 0,

    @Column(name = "ctacte")
    var cuentaCorriente: Byte = 0,

    var debita: Byte = 0,
    var iva: Byte = 0,
    var ganancias: Byte = 0,
    var aplicable: Byte = 0,

    @Column(name = "libroiva")
    var libroIva: Byte = 0,

    @Column(name = "tipocomprob")
    var letraComprobante: String? = null,

    var recibo: Byte = 0,
    var contado: Byte = 0,
    var transferencia: Byte = 0,
    var imprime: Byte = 0,

    @Column(name = "cgoanulacion")
    var comprobanteIdAnulacion: Int? = null,

    @Column(name = "cgocentroentrega")
    var centroStockIdEntrega: Int? = null,

    @Column(name = "cgocentrorecibe")
    var centroStockIdRecibe: Int? = null,

    @Column(name = "diasvigencia")
    var diasVigencia: Int? = null,

    var concepto: Byte = 0,
    var personal: Byte = 0,
    var comanda: Byte = 0,

    @Column(name = "tco_puntovta")
    var puntoVenta: Int = 0,

    @Column(name = "cgomozo")
    var codigoMozo: Byte = 0,

    @Column(name = "tco_transferir")
    var transferir: Byte = 0,

    @Column(name = "tco_cgocontable")
    var numeroCuenta: Long? = null,

    @Column(name = "tco_nivel")
    var nivel: Int = 0,

    @Column(name = "tco_fiscalizadora")
    var fiscalizadora: Byte = 0,

    @Column(name = "tco_invisible")
    var invisible: Byte = 0,

    @Column(name = "tco_tipoafip")
    var comprobanteAfipId: Int? = null,

    @Column(name = "tco_factelect")
    var facturaElectronica: Byte = 0,

    var asociado: Byte = 0,

    @OneToOne(optional = true)
    @JoinColumn(name = "tco_cgocontable", insertable = false, updatable = false)
    var cuenta: Cuenta? = null,

    @OneToOne(optional = true)
    @JoinColumn(name = "tco_tipoafip", insertable = false, updatable = false)
    var comprobanteAfip: ComprobanteAfip? = null

) : Auditable() {

    fun jsonify(): String {
        return Jsonifier.builder(this).build()
    }

}
